spec/integration/user_index_spec.rb

```diff
@@ -0,0 +1,33 @@
+require 'rails_helper'
+
+describe 'User Index Page Features', type: :feature, js: true do
+  before :each do
+    @user1 = User.create(
+      name: 'Harley Quinn',
+      photo: 'https://images.unsplash.com/photo-1696537768609-1cf03f53e893?ixlib=rb-4.0.3&ixid=M3wxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8fA%3D%3D&auto=format&fit=crop&w=1374&q=80',
+      bio: 'user1 Pix',
+      posts_counter: 5
+    )
+  end
+
+  it 'can see the username of all other users' do
+    visit users_path
+    expect(page).to have_all_of_selectors('.user-info')
+  end
+
+  it 'can see the profile pic of each users' do
+    visit users_path
+    expect(page).to have_all_of_selectors('.photo')
+  end
+
+  it 'can see the number of posts each user has written' do
+    visit users_path
+    expect(page).to have_all_of_selectors('.posts')
+  end
+
+  it 'redirect to the user\'s show page' do
+    visit users_path
+    click_link(href: user_path(@user1.id))
+    expect(page).to have_current_path(user_path(@user1.id))
```

Comment:
- [ ] I appreciate your huge efforts and I think you have some failing tests here so I highly recommend fixing them to be able to have all your features tested well so please make sure to fix them first ⭐💪🏻♻🔥

![image](https://github.com/rabayarabu/RoR-Blog-App/assets/58553711/2e76b21f-d29a-40eb-87ba-be2fa125b9cf)


- Here's some of failed examples
![image](https://github.com/rabayarabu/RoR-Blog-App/assets/58553711/af5b3f13-473f-470a-a055-45bb07fc8d2b)